packages/plugins/graphql/server/services/builders/filters/content-type.js

```diff
@@ -87,6 +92,22 @@ module.exports = ({ strapi }) => {
     builder.field(attributeName, { type: getFiltersInputTypeName(model) });
   };
 
+  const addComponentAttribute = (builder, attributeName, attribute) => {
```

Comment:
A small explanation for the code snippet:

Here we're only defining an input type for the filters of a given content type. Thus we don't need to add resolvers or args as you did below. We simply need to indicate the GraphQL type (let's call it T) of a component attribute in the filters.

This type T is already defined by calling the [registerFiltersDefinition](https://github.com/strapi/strapi/blob/graphql-nested-component-filters/packages/plugins/graphql/server/services/content-api/index.js#L157) for each content type AND every component in the [registerAPITypes](https://github.com/strapi/strapi/blob/graphql-nested-component-filters/packages/plugins/graphql/server/services/content-api/index.js#L147) method of the content-api service.

With that in mind, we can safely assume that there is a filter type (T) definition for every component and that we can reuse those types for our component attributes.

This is why instead of dealing with resolver logic or even defining new types here, all we have to do is to check if the component field is enabled and -if it's not- add its filter type (T)) to the parent type.